wfm.c

```diff
@@ -221,7 +221,7 @@ void checkfilename(char *inp_filename) {
 
     snprintf(temp_dirname, sizeof(temp_dirname), "%s", wp.phys_filename);
     if(strlen(dirname(temp_dirname)) < strlen(cfg.homedir)) 
-        error("Basename path too short");
+        error("Requested path basename ('%s') does not start with the homedir string ('%s'), assuming access is not allowed.");
```

Comment:
While I agree that the original error message is cryptic, this one is doubly so. Please think how to word it more concisely. Maybe just say "Access Denied basename(%s) != %s" or something along these lines. Also looks like you forgot to include arguments for both %s.